packages/expo-image/src/ExpoImage.tsx

```diff
@@ -17,7 +17,37 @@ export default function ExpoImage({ source, style, ...props }: ImageProps) {
   // additional layout passes, which we want to prevent.
   if (!Array.isArray(resolvedSource)) {
     const { width, height } = resolvedSource;
-    resolvedStyle = StyleSheet.flatten([{ width, height }, style]);
+    resolvedStyle.width = resolvedStyle.width ?? width;
+    resolvedStyle.height = resolvedStyle.height ?? height;
+  }
+
+  // Shadows behave different on iOS, Android & Web.
+  // Android uses the `elevation` prop, whereas iOS
+  // and web use the regular `shadow...` props.
+  let hasShadows = false;
+  if (Platform.OS === 'android') {
+    delete resolvedStyle.shadowColor;
+    delete resolvedStyle.shadowOffset;
+    delete resolvedStyle.shadowOpacity;
+    delete resolvedStyle.shadowRadius;
+    hasShadows = !!resolvedStyle.elevation;
+  } else {
+    delete resolvedStyle.elevation;
+    hasShadows = !!resolvedStyle.shadowColor;
+  }
+
+  // Shadows are rendered quite differently on iOS, Android and web.
+  // - iOS renders the shadow along the transparent contours of the image.
+  // - Android renders an underlay which extends to the inside of the bounds.
+  // - Web renders the shadow only on the outside of the bounds.
+  // To achieve a consistent appearance on all platforms, it is highly recommended
+  // to set a background-color on the Image when using shadows. This will ensure
+  // consistent rendering on all platforms and mitigate Androids drawing artefacts.
+  if (hasShadows && !resolvedStyle.backgroundColor) {
```

Comment:
Yes I think we should indeed do that. Initially I thought I'd allow people to specify transparent colors to suppress this warning altogether. Although this would then not render correctly on Android. It would render okay on iOS as iOS uses a pixel based shadow strategy. But it would be slowish.

But that thinking doesn't feel right anymore. I think we should indeed log the warning when a non-transparent background is set. This will guard the user from making mistakes that causes drawing artefacts on Android and potentially bad performance on iOS.

And if in the future we would like to support pixel based shadows on iOS then we can add a special prop for that (e.g. `shadowRenderMode`).

Reply:
I've updated the check so that it now always throws a warning when no- or a transparent background-color is provided 👍